### PR TITLE
feat: support for configuring the number of etcd health check retries

### PR DESCRIPTION
**Pull request by tylitianrui**
### What this PR does / why we need it:

related [5158](https://github.com/apache/apisix/pull/5158).   users can configure  the time of retry

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: conf/config-default.yaml [hunk 1]
```diff
@@ -274,6 +274,7 @@ etcd:
   timeout: 30                     # 30 seconds
   #resync_delay: 5                # when sync failed and a rest is needed, resync after the configured seconds plus 50% random jitter
   #health_check_timeout: 10       # etcd retry the unhealthy nodes after the configured seconds
+  health_check_retry: 2           # etcd retry time,defalut 2
```
Comment (tzssangglass) [inline conf/config-default.yaml:277]:
```suggestion
  health_check_retry: 2           # etcd retry time, defalut 2 
```

File: apisix/cli/etcd.lua [hunk 2]
```diff
@@ -196,7 +196,9 @@ function _M.init(env, args)
 
         local res, err
         local retry_time = 0
-        while retry_time < 2 do
+        
+        local health_check_retry = tonumber(yaml_conf.etcd.health_check_retry or  2)
```
Comment (tzssangglass) [inline apisix/cli/etcd.lua:200]:
```suggestion
        local health_check_retry = tonumber(yaml_conf.etcd.health_check_retry or 2)
```

File: apisix/cli/etcd.lua [hunk 3]
```diff
@@ -196,7 +196,9 @@ function _M.init(env, args)
 
         local res, err
         local retry_time = 0
-        while retry_time < 2 do
+        
+        local health_check_retry = tonumber(yaml_conf.etcd.health_check_retry or 2)
```
Comment (spacewander) [inline apisix/cli/etcd.lua:200]:
```suggestion
        local health_check_retry = tonumber(yaml_conf.etcd.health_check_retry) or 2
```

File: conf/config-default.yaml [hunk 4]
```diff
@@ -274,6 +274,7 @@ etcd:
   timeout: 30                     # 30 seconds
   #resync_delay: 5                # when sync failed and a rest is needed, resync after the configured seconds plus 50% random jitter
   #health_check_timeout: 10       # etcd retry the unhealthy nodes after the configured seconds
+  health_check_retry: 2           # etcd retry time, defalut 2
```
Comment (spacewander) [inline conf/config-default.yaml:277]: We should emphasize that the retry time only affects the health check in the comment.

Comment (tylitianrui) [reply conf/config-default.yaml:277]: thx, i will specify it

**Review by tzssangglass — DISMISSED** [review]

**Comment by tzssangglass** [comment]
hi @tylitianrui , PLAT: https://github.com/apache/apisix/runs/5210191215?check_suite_focus=true

**Comment by tylitianrui** [comment]
> hi @tylitianrui , PLAT: https://github.com/apache/apisix/runs/5210191215?check_suite_focus=true

thx, i will remove the whitespace

**Review by starsz — COMMENTED** [review]
Hi @tylitianrui.
It's better to add test case in t/cli/test_etcd.sh to cover it.

**Review by spacewander — APPROVED** [review]
Actually, the test case is in https://github.com/apache/apisix/blob/master/t/cli/test_etcd_healthcheck.sh.

I think it is too tough for the first contributor to add a test for it...

**Review by tokers — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**